src/decryption/decryption.ts

```diff
@@ -10,13 +10,55 @@ interface IEncryptionOptions {
   ad: string;
 }
 
-export async function decryptWithKey({
+export async function decryptStringWithKey({
   serialized,
   key,
 }: {
   serialized: string;
   key: string;
 }): Promise<string | null> {
+  return decryptWithKey(
+    {
+      serialized,
+      key,
+    },
+    'utf8'
+  );
+}
+
+export async function decryptBinaryWithKey({
+  serialized,
+  key,
+}: {
+  serialized: string;
+  key: string;
+}): Promise<string | null> {
+  return decryptWithKey(
+    {
+      serialized,
+      key,
+    },
+    'raw'
+  );
+}
+
+/**
+ * @deprecated This method should be replaced by
+ * decryptStringWithKey or decryptBinaryWithKey method.
+ * The default encoding is 'raw' is suitable for binaries and 1 byte UTF-8.
+ * string with greater than 2 bytes UTF-8 string will produce an incorrect result. e.g. data string '鍵键'
+ * encrypted with 'raw' encoding will produce incorrect decrypted value.
+ */
+export async function decryptWithKey(
+  {
+    serialized,
+    key,
+  }: {
+    serialized: string;
+    key: string;
+  },
+  encoding?: Encoding
```

Comment:
Should we provide a default encoding here of `raw` so the method still functions as it did before when upgrading?

Reply:
added default to 'raw' in all missing places